File: Python3/프로그래머스/0/181859. 배열 만들기 6/배열 만들기 6.py
def solution(arr):
    answer = []
    
    for i in range(len(arr)):
        if len(answer) == 0 or answer[::-1][0] != arr[i]:
            answer.append(arr[i])
        else:
            answer.pop()
    return answer if len(answer) > 0 else [-1]

File: C++/프로그래머스/0/181859. 배열 만들기 6/배열 만들기 6.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<int> arr) {
    vector<int> answer;
    for(int i = 0; i < arr.size();i++) {
        if(answer.empty() || answer.back() != arr[i]) {
            answer.push_back(arr[i]);
        } else {
            answer.pop_back();
        }
    }
    
    if(answer.empty()) answer.push_back(-1);
    return answer;
}